### PR TITLE
Fix various bugs in Atomix agent configuration

### PR DESCRIPTION
**Pull request by kuujo**
This PR fixes various bugs in the agent configuration.

### CODE REVIEW

File: agent/src/main/java/io/atomix/agent/AtomixAgent.java [hunk 1]
```diff
@@ -78,13 +76,13 @@ public static void main(String[] args) throws Exception {
...
         .nargs("*")
-        .type(nodeArgumentType)
+        .type(nodeArgumentType.apply(Node.Type.CORE))
         .metavar("NAME:HOST:PORT")
         .required(false)
         .help("Sets the core cluster configuration");
     parser.addArgument("--bootstrap-nodes", "-b")
         .nargs("*")
-        .type(nodeArgumentType)
+        .type(nodeArgumentType.apply(Node.Type.DATA))
```
Comment (johnou) [inline agent/src/main/java/io/atomix/agent/AtomixAgent.java]: data for bootstrap nodes? I haven't used args4j much..

Comment (kuujo) [reply agent/src/main/java/io/atomix/agent/AtomixAgent.java]: If `--core-nodes` are provided, they'll be used as bootstrap nodes, so the assumption is that any `--bootstrap-nodes` provided will be `DATA` nodes

Comment (johnou) [reply agent/src/main/java/io/atomix/agent/AtomixAgent.java]: what if they aren't? is it corrected with gossip?

Comment (kuujo) [reply agent/src/main/java/io/atomix/agent/AtomixAgent.java]: if `--core-nodes` aren't provided then presumably there _are_ no core nodes. This is an unavoidable limitation of consensus. A static consensus configuration must be provided or correctness is risked.

Comment (johnou) [reply agent/src/main/java/io/atomix/agent/AtomixAgent.java]: I mean what if I start a node as core and configure that node as data in another node.

Comment (kuujo) [reply agent/src/main/java/io/atomix/agent/AtomixAgent.java]: The problem is, if different nodes are started with different sets of `CORE` nodes, they'll count votes differently and can thus elect multiple leaders, leading to split brain. The CORE configuration must be the same on all nodes at startup. There's no way around that limitation. A single source must be assumed - either in a provided configuration or by querying a single external source.

**Pull request merged**